src/mono/wasm/Wasm.Build.Tests/Blazor/IcuShardingTests.cs

```diff
@@ -0,0 +1,116 @@
+using Xunit.Abstractions;
+using Xunit.Sdk;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+#nullable enable
+
+namespace Wasm.Build.Tests.Blazor;
+
+// these tests only check if correct ICU files got copied
+public class IcuShardingTests : BlazorWasmTestBase
+{
+    public IcuShardingTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+        : base(output, buildContext) {}
+
+    [Theory]
+    [InlineData("Debug", "icudt.dat")]
+    [InlineData("Release", "icudt.dat")]
+    [InlineData("Debug", "icudt_EFIGS.dat")]
+    [InlineData("Release", "icudt_EFIGS.dat")]
+    [InlineData("Debug", "icudt_no_CJK.dat")]
+    [InlineData("Release", "icudt_no_CJK.dat")]
+    [InlineData("Debug", "icudt_CJK.dat")]
+    [InlineData("Release", "icudt_CJK.dat")]
+    public async Task CustomIcuFileFromRuntimePack(string config, string fileName)
+    {
+        string id = $"blz_customFromRuntimePack_{config}_{GetRandomId()}";
+        string projectFile = CreateBlazorWasmTemplateProject(id);
+        var buildOptions = new BlazorBuildOptions(
+                id,
+                config,
+                WarnAsError: true,
+                GlobalizationMode: GlobalizationMode.PredefinedIcu,
+                PredefinedIcudt: fileName
+            );
+        AddItemsPropertiesToProject(
+            projectFile,
+            extraProperties: 
+                $"<BlazorIcuDataFileName>{fileName}</BlazorIcuDataFileName>");
+
+        (CommandResult res, string logPath) = BlazorBuild(buildOptions);
+        await BlazorRunForBuildWithDotnetRun(new BlazorRunOptions() { Config = config });
+    }
+
+    [Theory]
+    [InlineData("Debug", "incorrectName.dat", false)]
+    [InlineData("Release", "incorrectName.dat", false)]
+    [InlineData("Debug", "icudtNonExisting.dat", true)]
+    [InlineData("Release", "icudtNonExisting.dat", true)]
+    public void NonExistingCustomFileAssertError(string config, string fileName, bool isFilenameCorrect)
+    {
+        string id = $"blz_invalidCustomIcu_{config}_{GetRandomId()}";
+        string projectFile = CreateBlazorWasmTemplateProject(id);
+        AddItemsPropertiesToProject(
+            projectFile,
+            extraProperties: 
+                $"<BlazorIcuDataFileName>{fileName}</BlazorIcuDataFileName>");
+
+        try
+        {
+            (CommandResult res, string logPath) = BlazorBuild(
+                new BlazorBuildOptions(
+                    id,
+                    config,
+                    WarnAsError: false,
+                    GlobalizationMode: GlobalizationMode.PredefinedIcu,
+                    PredefinedIcudt: fileName
+                ));
+        }
+        catch (XunitException ex)
+        {
+            if (isFilenameCorrect)
+            {
+                Assert.Contains($"File in location $(BlazorIcuDataFileName)={fileName} cannot be found: in project directory, in runtime pack nor used as an absolute path.", ex.Message);
+            }
```

Comment:
I don't like this, but I also see that there is not a good way to do it right now either. I will fix that later.

Reply:
You mean the way we test it or the way we behave when the file is not found?